solr/core/src/java/org/apache/solr/handler/admin/api/ZookeeperAPI.java

```diff
@@ -0,0 +1,285 @@
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import org.apache.solr.api.JerseyResource;
+import org.apache.solr.cloud.ZkController;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrException.ErrorCode;
+import org.apache.solr.common.cloud.SolrZkClient;
+import org.apache.solr.common.cloud.ZkDynamicConfig;
+import org.apache.solr.common.params.SolrParams;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.handler.admin.ZookeeperInfoHandler.FilterType;
+import org.apache.solr.handler.admin.ZookeeperInfoHandler.PageOfCollections;
+import org.apache.solr.handler.admin.ZookeeperInfoHandler.PagedCollectionSupport;
+import org.apache.solr.handler.admin.ZookeeperInfoHandler.ZKPrinter;
+import org.apache.solr.handler.admin.ZookeeperStatusHandler;
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/cluster/zookeeper/")
+public class ZookeeperAPI extends JerseyResource {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final CoreContainer coreContainer;
+  private final SolrQueryRequest solrQueryRequest;
+  private final SolrQueryResponse solrQueryResponse;
+  private PagedCollectionSupport pagingSupport;
+
+  @Inject
+  public ZookeeperAPI(
+      CoreContainer coreContainer,
+      SolrQueryRequest solrQueryRequest,
+      SolrQueryResponse solrQueryResponse) {
+    this.coreContainer = coreContainer;
+    this.solrQueryRequest = solrQueryRequest;
+    this.solrQueryResponse = solrQueryResponse;
+  }
+
+  @GET
+  @Path("/files")
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @Operation(
+      summary = "List Zookeeper files.",
+      tags = {"zookeeperFiles"})
+  @PermissionName(ZK_READ_PERM)
+  public ZookeeperFilesResponse getFiles() throws Exception {
+    final ZookeeperFilesResponse response = instantiateJerseyResponse(ZookeeperFilesResponse.class);
+    final SolrParams params = solrQueryRequest.getParams();
+    Map<String, String> map = new HashMap<>(1);
+    map.put(WT, "raw");
+    map.put(OMIT_HEADER, "true");
+    // solrQueryRequest.setParams(SolrParams.wrapDefaults(new MapSolrParams(map), params));
+    synchronized (this) {
+      if (pagingSupport == null) {
+        pagingSupport = new PagedCollectionSupport();
+        ZkController zkController = coreContainer.getZkController();
+        if (zkController != null) {
+          // get notified when the ZK session expires (so we can clear the cached collections and
+          // rebuild)
+          zkController.addOnReconnectListener(pagingSupport);
+        }
+      }
+    }
+
+    String path = params.get(PATH);
```

Comment:
[-1] Fetching parameters off of the `SolrQueryRequest` gets the job done, but it doesn't make use of the strong-typing or clear enumeration of inputs/outputs that's one of the big draws of using JAX-RS for our APIs.

i.e. It makes it hard to tell what the APIs inputs are from the method signature alone.  Any OpenAPI spec we generate for this API won't know about the stuff we get off of SolrQueryRequest, etc.

I'd rather we put any inputs that this API requires up in the method signature (and modify the v1 code to pass those in when it calls `ZooKeeperAPI.getFile`